lib/formatter/index.js

```diff
@@ -64,16 +64,24 @@ Formatter.prototype.unformat = function (options) {
   var start = options.selection.start;
   var end = options.selection.end;
   var unformattedString = '';
+  var formatCompensation = 0;
 
   for (i = 0; i < this.pattern.length; i++) {
     patternChar = this.pattern[i];
 
-    if (!patternChar.isPermaChar && options.value[i] != null && patternChar.value.test(options.value[i])) {
-      unformattedString += options.value[i];
+    if (
```

Comment:
This logic is getting gnarly. Let's pull it out into it's own function.

Reply:
passed into function `isPatterntChar`